pact/Pact/Core/IR/Eval/CEK/Types.hs

```diff
@@ -228,134 +228,134 @@ data CapTokenClosure i
+pattern VDefClosure :: Closure e step b i -> CEKValue e step b i
 pattern VDefClosure clo = VClosure (C clo)
 
-pattern VLamClosure :: LamClosure step b i m -> CEKValue step b i m
+pattern VLamClosure :: LamClosure e step b i -> CEKValue e step b i
 pattern VLamClosure clo = VClosure (LC clo)
 
-pattern VPartialClosure :: PartialClosure step b i m -> CEKValue step b i m
+pattern VPartialClosure :: PartialClosure e step b i -> CEKValue e step b i
 pattern VPartialClosure clo = VClosure (PC clo)
 
-pattern VDefPactClosure :: DefPactClosure step b i m -> CEKValue step b i m
+pattern VDefPactClosure :: DefPactClosure e step b i -> CEKValue e step b i
 pattern VDefPactClosure clo = VClosure (DPC clo)
 
 -- | Result of an evaluation step, either a CEK value or an error.
-data EvalResult (step :: CEKStepKind) (b :: K.Type) (i :: K.Type) (m :: K.Type -> K.Type)
-  = EvalValue (CEKValue step b i m)
+data EvalResult (e :: RuntimeMode) (step :: CEKStepKind) (b :: K.Type) (i :: K.Type)
+  = EvalValue (CEKValue e step b i)
   | VError [StackFrame i] UserRecoverableError i
   deriving (Show, Generic)
 
-instance (NFData b, NFData i) => NFData (EvalResult step b i m)
+instance (NFData b, NFData i) => NFData (EvalResult e step b i)
 
 
 
 
-type NativeFunction (step :: CEKStepKind) (b :: K.Type) (i :: K.Type) (m :: K.Type -> K.Type)
-  = i -> b -> Cont step b i m -> CEKErrorHandler step b i m -> CEKEnv step b i m -> [CEKValue step b i m] -> m (CEKEvalResult step b i m)
+type NativeFunction (e :: RuntimeMode) (step :: CEKStepKind) (b :: K.Type) (i :: K.Type)
+  = i -> b -> Cont e step b i -> CEKErrorHandler e step b i -> CEKEnv e step b i -> [CEKValue e step b i] -> EvalM e b i (CEKEvalResult e step b i)
 
-data NativeFn (step :: CEKStepKind) (b :: K.Type) (i :: K.Type) (m :: K.Type -> K.Type)
+data NativeFn (e :: RuntimeMode) (step :: CEKStepKind) (b :: K.Type) (i :: K.Type)
   = NativeFn
   { _native :: !b
-  , _nativeEnv :: !(CEKEnv step b i m)
-  , _nativeFn :: !(NativeFunction step b i m)
+  , _nativeEnv :: !(CEKEnv e step b i)
+  , _nativeFn :: !(NativeFunction e step b i)
   , _nativeArity :: {-# UNPACK #-} !Int
   , _nativeLoc :: !i
   } deriving (Generic)
 
-instance (NFData b, NFData i) => NFData (NativeFn step b i m)
+instance (NFData b, NFData i) => NFData (NativeFn e step b i)
 
 -- | A partially applied native because we don't allow
 -- them to be applied at the lhs of an app since pact historically hasn't had partial closures.
 -- This is a bit annoying to deal with but helps preserve semantics
-data PartialNativeFn (step :: CEKStepKind) (b :: K.Type) (i :: K.Type) (m :: K.Type -> K.Type)
+data PartialNativeFn (e :: RuntimeMode) (step :: CEKStepKind) (b :: K.Type) (i :: K.Type)
   = PartialNativeFn
   { _pNative :: !b
-  , _pNativeEnv :: !(CEKEnv step b i m)
-  , _pNativeFn :: !(NativeFunction step b i m)
+  , _pNativeEnv :: !(CEKEnv e step b i)
+  , _pNativeFn :: !(NativeFunction e step b i)
   , _pNativeArity :: {-# UNPACK #-} !Int
```

Comment:
Should we remove the pragma too?

Reply:
Good point. It's pretty redundant, so yes